### ISSUE TITLE
JPEG decoding changes the image contrast and/or lacks minor details

### ISSUE THREAD

**Issue by artlav** [intel/media-driver#1193]
Tried on:

    i5-6260U (Iris Graphics 540)
    m5-6Y54 (HD Graphics 515)
    i3-10110U (UHD Graphics)

Not sure if it's an issue or a limitation on the decoding.

Take loadjpeg form libva-utils, run it on any file with high contrast and/or fine details, compare to software decoding.

What is expected: https://orbides.org/img/ref.png

What is produced: https://orbides.org/img/res.png

JPG file parameters do not appear to matter, every single file i tried exhibit this effect to some degree.

I was not able to google if the hardware decode is supposed to be this imprecise or if there are some color space correction settings that need to be set or something, so i file this as an issue for now.

**Comment by wangyan-intel** [comment]
@Jexu Could you please take a look? Thanks.

**Comment by XinfengZhang** [comment]
hi @artlav  I suppose it is not related with JPEG decoding itself. as you know, there are different format jpeg. the format of jpeg and output surface maybe is different. so maybe sometime there are a VPP followed 
for example:  the jpeg is 444 or 422Y/422H, but application create a NV12 surface for the output. then a internal color format conversion was called. some details left.
so, could you help to check the color format of jpeg and output surface . or share the reproducer?

**Comment by XinfengZhang** [comment]
of course, it also maybe related with Qmatrix (value and layout) , so, still need reproducer :)

**Comment by artlav** [comment]
Problem is, EVERY jpg file is off like that, which is what makes me wonder if there is something wrong with loadjpeg form libva-utils. It does create the output surface to match the input file.

Samples:
420: https://orbides.org/img/420.jpg

422V: https://orbides.org/img/422.jpg

444: https://orbides.org/img/444.jpg

All decode to: https://orbides.org/img/bad.png

Should look like: https://orbides.org/img/good.png (i.e. in browser)

Another observation is that HW decode and encode are mutually consistent in their wrongness - jpegenc output would appear brighter for an SW decoder, but correct for loadjpeg.

**Comment by artlav** [comment]
This is still an issue as of 22.4.3, and is still perfectly reproducible every time for every image.

Can also add i7-1195G7 to the list of CPUs i tried this on.

**Comment by Jexu** [comment]
This maybe related with qmatrix, and it can get right output in bellow picture by raising the value of qmatrix. But hard to say it is hw limitation or something others since the qmatrix is provided by uplayer app/lib.
![image](https://user-images.githubusercontent.com/11835821/177100719-37afcee3-6363-47f1-9dc9-b8494664983e.png)

**Comment by artlav** [comment]
The iq_matrix.quantiser_table is copied over from the JPG file, verbatim. Not sure what else the lib can do.

I tried to tweak it as well, and it's too dim with +1 and too bright with -1. Almost feels like it have to be +0.5, but there is no way to make it so.

**Comment by Jexu** [comment]
So, may i know any impacts for you with the hw result?

**Comment by artlav** [comment]
I'm working on an image viewer. HW decoding of JPG files is several times faster than software, which matters at DSLR camera image sizes. However i can not keep it on by default, since all images come out with the gamma visibly off.
